src/cpp/core/DateTime.cpp

```diff
@@ -96,6 +100,63 @@ std::string millisecondsSinceEpochAsString(double ms)
 
    return date_time::format(time, "%d %b %Y %H:%M:%S");
 }
+
+bool parseUtcTimeFromIsoString(const std::string& timeStr,
+                               boost::posix_time::ptime *pOutTime)
+{
+   return parseUtcTimeFromFormatString(timeStr,
+                                       "%Y-%m-%d %H:%M:%S %ZP",
+                                       pOutTime);
+}
+
+bool parseUtcTimeFromFormatString(const std::string& timeStr,
+                                  const std::string& formatStr,
+                                  boost::posix_time::ptime *pOutTime)
+{
+   using namespace boost::local_time;
+
+   std::stringstream ss(timeStr);
+   local_time_input_facet* ifc = new local_time_input_facet(formatStr);
```

Comment:
It does not leak. The facets are managed by the stream per @jjallaire. This is just a quirkiness of the library here. If you free it you will get a crash due to double free.